Doc: purchase ckb on exchanges (#105)
godwokenrises/godwoken-doc: docs/purchaseCkb.md

- [Deposit-lock](https://github.com/godwokenrises/godwoken/blob/ad898e7a9821df3e627737d32865bec85bf1a753/docs/deposit_and_withdrawal.md#deposit): The lock is for depositing assets from CKB L1 to Godwoken L2, the block producer of Godwoken will collect cells with deposit-lock and process the deposit requests in blocks.

If the above content is still vague to you, perhaps think of it this way: Different locks correspond to different L1 addresses, and some exchanges just don’t reorganize locks other than the `SECP256K1/blake160` lock. Therefore `L1 Wallet Address` (Omni-lock) and `L1 Deposit Address` (Deposit-lock) may not be valid in the eyes of these exchanges.

### How to solve it

If you have encountered this issue, you can follow the below instructions to safely transfer your CKB to the `L1 Wallet Address` or the `L1 Deposit Address` of the target account.

> Since the problem we’re facing is that the exchange don’t support the type of address we tend to transfer to, the overall solution is to add an extra step in the middle of buying CKB. Let’s take `L1 Wallet Address` as an example. 

Reviewer: ```suggestion
Since the problem we’re facing is that the exchange doesn’t support the type of address we tend to transfer to, the overall solution is to add an extra step in the middle of buying CKB. Let’s take `L1 Wallet Address` as an example. 
```

The purchase process previously:

```
Purchase CKB on the exchange → Withdraw to L1 Wallet Address
```

New purchase process that solves our problem:

```
Purchase CKB on the exchange → **Withdraw to a crypto wallet** → Transfer to L1 Wallet Address
```

You can see the solution is to withdraw to a crypto wallet that supports CKB, and then transfer to the `L1 Wallet Address` of the target account. This is because crypto wallets usually has a higher compatibility to different locks (or they just don't verify the lock of the recipient’s address).

> This solution should be simple enough, but it also has a disadvantage: you always need to leave some CKB in your crypto wallet for paying transaction fees. So in this solution, you will have to leave some CKB in your exchange’s wallet, and leave some CKB in your crypto wallet as well. However, this is a solid solution to the problem we’re facing.

Author: ```suggestion
This solution should be simple enough, but as it always need us to make one more step in the process of purchasing CKB, the best solution for us should be to purchase on an exchange that allows us to withdraw to different kind of addresses. However, this is still a solid solution to the problem we’re facing.
```

---
id: purchaseCKB
title: Purchase CKB
---
import useBaseUrl from "@docusaurus/useBaseUrl";

## Purchase and withdraw

### Purchase on an exchange

If you need to use CKB/pCKB on mainnet, the most common way to get it is to purchase it from an exchange. There are a lot of exchanges on the market, you can visit here to pick an exchange to purchase CKB: [CKB market-cap and chart | CoinMarketCap](https://coinmarketcap.com/currencies/nervos-network/markets/).

The general steps to purchase CKB should be simple, but you may still need some guidance when you complete the purchase and want to transfer your CKB to CKB L1 or Godwoken L2. 

If so, please continue to read the following sections.

:::tip

> After browsing the CoinMarketCap site, if you still don't know what exchange to use, here are some of the exchanges we tested, which definitely support the withdrawal of CKB to your `L1 wallet address` and `L1 recharge address`:

Reviewer 2: ```suggestion
After browsing the CoinMarketCap site, if you still don't know which exchange to use, here are some of the exchanges we tested, which definitely support the withdrawal of CKB to your `L1 Wallet Address` or `L1 Deposit Address`:
```

Reviewer 2: unified the name in the context of this article.

others LGTM
